components/camel-zipfile/src/main/java/org/apache/camel/processor/aggregate/zipfile/ZipAggregationStrategy.java

```diff
@@ -178,81 +191,59 @@ public Exchange aggregate(Exchange oldExchange, Exchange newExchange) {
-        }
-        byte[] buffer = new byte[8192];
 
-        FileInputStream fis = new FileInputStream(tmpZip);
-        ZipInputStream zin = new ZipInputStream(fis);
-        ZipOutputStream out = new ZipOutputStream(new FileOutputStream(source));
-
-        try {
-            InputStream in = new FileInputStream(file);
-            out.putNextEntry(new ZipEntry(fileName == null ? file.getName() : fileName));
-            for (int read = in.read(buffer); read > -1; read = in.read(buffer)) {
-                out.write(buffer, 0, read);
+    private static void newZipFile(File zipFile) throws URISyntaxException, IOException {
+        if (zipFile.exists()) {
+            if (!zipFile.delete()) { // Delete, because ZipFileSystem needs to create file on its own (with correct END bytes in the file)
+                throw new IOException("Cannot delete file " + zipFile);
             }
-            out.closeEntry();
-            IOHelper.close(in);
+        }
+        Map<String, String> env = new HashMap<>();
+        env.put("create", Boolean.TRUE.toString());
 
-            for (ZipEntry ze = zin.getNextEntry(); ze != null; ze = zin.getNextEntry()) {
-                out.putNextEntry(ze);
-                for (int read = zin.read(buffer); read > -1; read = zin.read(buffer)) {
-                    out.write(buffer, 0, read);
-                }
-                out.closeEntry();
-            }
-        } finally {
-            IOHelper.close(fis, zin, out);
+        try (FileSystem ignored = FileSystems.newFileSystem(getZipURI(zipFile), env)) {
+            //noop, just open and close FileSystem to initialize correct headers in file
         }
-        tmpZip.delete();
     }
-    
-    private static void addEntryToZip(File source, String entryName, byte[] buffer, int length) throws IOException {
-        File tmpZip = Files.createTempFile(source.getName(), null).toFile();
-        tmpZip.delete();
-        if (!source.renameTo(tmpZip)) {
-            throw new IOException("Cannot create temp file: " + source.getName());
-        }
 
-        FileInputStream fis = new FileInputStream(tmpZip);
-        ZipInputStream zin = new ZipInputStream(fis);
-        ZipOutputStream out = new ZipOutputStream(new FileOutputStream(source));
-        try {
-            out.putNextEntry(new ZipEntry(entryName));
-            out.write(buffer, 0, length);
-            out.closeEntry();
+    private void addFileToZip(File zipFile, File file, String fileName) throws IOException, URISyntaxException {
+        String entryName = fileName == null ? file.getName() : fileName;
+        Map<String, String> env = new HashMap<>();
+        env.put("useTempFile", Boolean.toString(this.useTempFile));
+        try (FileSystem fs = FileSystems.newFileSystem(getZipURI(zipFile), env)) {
+            Path dest = fs.getPath("/", entryName);
+            Files.createDirectories(dest.getParent());
+            Files.copy(file.toPath(), dest, StandardCopyOption.REPLACE_EXISTING);
```

Comment:
this is `REPLACE_EXISTING`, the other one is `CREATE`
and only charset is the extra piece. wondering if this could be simplified to save a couple of more lines.

Reply:
In addition to charset. The one method is working with `Files.write` from buffer parameter and the other one `Files.copy` with file parameter. Refactoring to single method could be IMHO ugly. 

You are right with Standard(Copy|Open)Option. It makes sense to have consistent behavior in the two method, I will replace `StandardOpenOption.CREATE` with `StandardOpenOption.TRUNCATE_EXISTING`

Reply:
`TRUNCATE_EXISTING` added in the last commit